chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-attack-vectors-of-lightning-network.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Fabrice Drouin - Attack Vectors of Lightning Network (2019-06-25)
+transcript_by: Gijs van Dam
+categories: ['residency']
+tag: ['attacks', 'lightning']
+---
+
+Name: Fabrice Drouin
+
+Topic: Attack Vectors of Lightning Network
+
+Location: Chaincode Residency – Summer 2019
+
+Date: June 25th 2019
+
+Video: https://youtu.be/R5cSrftd8nc
+
+# Introduction
+
+All right so I'm going to introduce really quickly attack vectors on Lightning. I focus on first what you can do with the Lightning protocol, but I will mostly speak about how attacks will probably happen in real life. It's probably not going to be direct attacks on the protocol itself. 
+
+# Denial of Service
+
+So the basic attacks you can have when you're running lightning nodes are denial of service attacks basically. Lightning nodes are servers that accept incoming tcp connections, you can just overflow the servers with connections and that's something that is extremely hard to fight. Basically fighting low level connection DDoS  is really expensive. You can use services like Cloudflare but it's very expensive and if you don't what's going to happen is your host your hosting provider will eventually disconnect you from the network there's no route to you because DDoS attack against you will have an impact on everyone else in the same data centers and you will be effectively disconnected so that's really really hard to fight. 
+
+There's an ongoing DDos attack  against electrum servers, I don’t know if you have heard of it. They are blacklisting I think 150,000 IPs right now, but it's a battle that is really hard to fight. Something that is really easy to do with lightning but it's a bit costly is to just lock up channels. You have limitations in lightning. You can't have more than, you can’t have too many pending payments and the amounts of your pending payments is also limited, so you can just lock up channels. You can just send HTLC’s without pre-images and after a while channels will not accept more pending payments. However this is not free. So can you tell me why it's not really free to do that? 

```

Comment:
```suggestion
There's an ongoing DDos attack against electrum servers, I don’t know if you have heard of it. They are blacklisting I think 150,000 IPs right now, but it's a battle that is really hard to fight. Something that is really easy to do with lightning but it's a bit costly is to just lock up channels. You have limitations in lightning. You can't have more than, you can’t have too many pending payments and the amounts of your pending payments is also limited, so you can just lock up channels. You can just send HTLC’s without pre-images and after a while, channels will not accept more pending payments. However, this is not free. So can you tell me why it's not really free to do that? 

```

Also, the use of preimage is inconsistent. There are 6 preimages and 2 pre-images.